docs/adr/0000-database-upgrade-testing.md

```diff
@@ -0,0 +1,60 @@
+# Database Upgrade Testing
+
+* Status: accepted
+* Deciders: [Sander, Arnaud, Wouter]
+* Date: 2023-01-12
+
+## Context and Problem Statement
+
+When making updates to the database schema, it is important to test this.
+
+## Decision Drivers
+
+* We want to test that database update code correctly transforms data produced by older code to data that can be handled by the current code
+* But within the code base, this older code no longer exists
```

Comment:
```suggestion
* But within the code base, this older code no longer exists or has been modified, so it can not be relied upon in the test case before the migration, where database schema and code might be out of sync.
```
Part of https://github.com/inmanta/inmanta-core/pull/5362/files#r1069134210